Implement CM Janus synchronization

fvovan opened the pull request:
Implement synchronize method on CM_Janus_Service which uses cm-janus status to add/remove channels and their streams.
This method should be called within maintenance cli when service configured.

fvovan commented on library/CM/Janus/Service.php:
```diff
@@ -20,7 +20,64 @@ public function __construct(CM_Janus_Configuration $configuration, CM_Janus_Http
+                    $streamRepository->removeStream($streamPublish);
+                }
+            } else {
+                //TODO Is it possible to have channel w/o publishers?
```
@tomaszdurka wdyt?

fvovan commented:
@tomaszdurka  please review,
generally it works but i'm thinking about edge cases
also we need either change output format in `/status` API method at cm-janus
or convert corresponding param names in CM's JanusHttpClient

tomaszdurka commented on library/CM/Janus/Service.php:
```diff
@@ -20,7 +20,64 @@ public function __construct(CM_Janus_Configuration $configuration, CM_Janus_Http
+        $startStampLimit = time() - 3;
+        $streamList = [];
+        $channelSubscriberList = [];
+        foreach ($this->_configuration->getServers() as $server) {
```
Oh I see you've copied synchronization from wowza. I think you can rather use `CM_Janus_Service-> _fetchStatus()`

fvovan replied:
Actually, not entirely. :) Yes, I saw this method, but as you can see there is some processing inside the loop.
I can fetch status using existing method but anyway will have to iterate through it once more.

fvovan commented:
it's adjusted to use `_fetchStatus`

tomaszdurka commented:
@fvovan can you double-check what we wrote, plus run tests with coverage.

tomaszdurka commented:
Change `CM_Janus_HttpApiClient` to expect `{id: stream.id, channelName: stream.channelName, isPublish: stream.isPublish};` response.

fvovan commented:
@tomaszdurka please review
I fixed some flaws and added several test cases.
Line coverage was increased from 84% to  96%  (basically only getConfigure() left untested, but it is very trivial)

tomaszdurka commented on library/CM/Janus/Service.php:
```diff
@@ -20,7 +20,55 @@ public function __construct(CM_Janus_Configuration $configuration, CM_Janus_Http
+        }
+
+        $startStampLimit = time() - 3;
+
```
Remove empty line?

tomaszdurka replied:
Oh this is on purpose I can see. For each block. Maybe rather not? Probably newline separator for separating stopping client and removing them from database is enough.

tomaszdurka commented:
Looks good except maybe those newlines. I've restarted failing build.

tomaszdurka commented:
@fvovan oh, you forgot to add it to maintenance cli.

Please add here https://github.com/cargomedia/cm/issues/1986.
Thanks!
